File: snippets/fsharp/VS_Snippets_CLR_System/System.ArgumentOutOfRangeException/fs/EmptyString1.fs
module EmptyString1

// <Snippet15>
open System

let getFirstCharacter (s: string) = s[0]

let words = [ "the"; "today"; "tomorrow"; " "; "" ]
for word in words do
    printfn $"First character of '{word}': '{getFirstCharacter word}'"

// The example displays the following output:
//    First character of 'the': 't'
//    First character of 'today': 't'
//    First character of 'tomorrow': 't'
//    First character of ' ': ' '
//
//    Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
//       at <StartupCode$argumentoutofrangeexception>.$EmptyString1.main@()
// </Snippet15>

module StringLib =
   // <Snippet16>
   let getFirstCharacter (s: string) =
      if String.IsNullOrEmpty s then
         '\u0000'
      else
         s[0]
   // </Snippet16>

File: snippets/csharp/System/ArgumentOutOfRangeException/Overview/EmptyString1.cs
// <Snippet15>
using System;

public class Example1
{
    public static void Main()
    {
        string[] words = { "the", "today", "tomorrow", " ", "" };
        foreach (string word in words)
            Console.WriteLine($"First character of '{word}': '{GetFirstCharacter(word)}'");
    }

    private static char GetFirstCharacter(string s) => s[0];
}
// The example displays the following output:
//    First character of //the//: //t//
//    First character of //today//: //t//
//    First character of //tomorrow//: //t//
//    First character of // //: // //
//
//    Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
//       at Example.Main()
// </Snippet15>

public static class StringLib
{
    // <Snippet16>
    static char GetFirstCharacter(string s)
    {
        if (string.IsNullOrEmpty(s))
            return '\u0000';
        else
            return s[0];
    }
    // </Snippet16>

    public static char DoNothing(string s) => GetFirstCharacter(s);
}
